plugin/pulpcore/plugin/stages/artifact_stages.py

```diff
@@ -174,17 +168,14 @@ def _downloaders_for_content(self, content):
         """
         Compute a list of downloader coroutines, one for each artifact to download for `content`.
 
-        When run, the downloader coroutine needs to get the download semaphore before downloading.
-
         Returns:
             List of downloader coroutines (may be empty)
         """
-        async def download_with_limit(semaphore, downloader):
-            async with semaphore:
-                log.debug("ArtifactDownloader: Start download of '%s'", declarative_artifact.url)
-                result = await downloader.run()
-                log.debug("ArtifactDownloader: Downloaded: '%s'", declarative_artifact.url)
-                return result
+        async def download_and_log(downloader):
+            log.debug("ArtifactDownloader: Start download of '%s'", declarative_artifact.url)
```

Comment:
I am wondering whether these log entries still make sense: If the stage work on 200 concurrent content units and 5 connections are allowed, the user will see 200 "Start download of..." logs, but only 5 of those will actually be downloading at a time. With the new approach, only the downloader may know what is actually downloading. Should we get rid of these messages here? Alternatively, we could change the message to "scheduling download..." or so.

Reply:
That is a good point about the logs not being exactly correct . I'm +1 on getting rid of the messages here, but if you do that the method does nothing which means I'm +1 on removing it altogether.

Reply:
Yes, I agree.